pkg/kubelet/container/helpers.go

```diff
@@ -110,28 +110,18 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 // Note: remember to update hashValues in container_hash_test.go as well.
 func HashContainer(container *v1.Container) uint64 {
 	hash := fnv.New32a()
-	// Omit nil or empty field when calculating hash value
-	// Please see https://github.com/kubernetes/kubernetes/issues/53644
-	containerJSON, _ := json.Marshal(container)
+	containerJSON, _ := json.Marshal(pickFieldToHash(container))
 	hashutil.DeepHashObject(hash, containerJSON)
 	return uint64(hash.Sum32())
 }
 
-// HashContainerWithoutResources returns the hash of the container with Resources field zero'd out.
-func HashContainerWithoutResources(container *v1.Container) uint64 {
-	// InPlacePodVerticalScaling enables mutable Resources field.
-	// Changes to this field may not require container restart depending on policy.
-	// Compute hash over fields besides the Resources field
-	// NOTE: This is needed during alpha and beta so that containers using Resources but
-	//       not subject to In-place resize are not unexpectedly restarted when
-	//       InPlacePodVerticalScaling feature-gate is toggled.
-	//TODO(vinaykul,InPlacePodVerticalScaling): Remove this in GA+1 and make HashContainerWithoutResources to become Hash.
-	hashWithoutResources := fnv.New32a()
-	containerCopy := container.DeepCopy()
-	containerCopy.Resources = v1.ResourceRequirements{}
-	containerJSON, _ := json.Marshal(containerCopy)
-	hashutil.DeepHashObject(hashWithoutResources, containerJSON)
-	return uint64(hashWithoutResources.Sum32())
+// pickFieldToHash pick fields that will affect the running status of the container for hash,
+// currently this field range only contains `image`.
+func pickFieldToHash(container *v1.Container) *v1.Container {
+	return &v1.Container{
```

Comment:
I have a slight preference for not relying on v1.Container marshaling... can this shift to accumulate into a `map[string]string` we can easily see what will marshal? That means if a new field shows up in Container that marshals by default, it won't destabilize our hash across minor versions

```go
func pickFieldToHash(container *v1.Container) map[string]any {
  retval := map[string]string{
    "name": container.Name,
    "image": container.Image,
  }
  return retval
}
```